Modules/Devices/src/Devices.Application/Devices/Commands/DeleteDevice/Handler.cs

```diff
@@ -22,17 +23,9 @@ public Handler(IIdentitiesRepository identitiesRepository, IUserContext userCont
     public async Task Handle(DeleteDeviceCommand request, CancellationToken cancellationToken)
     {
         var deviceThatIsDeletingId = _userContext.GetDeviceId();
-        var deviceThatIsDeleting = await _identitiesRepository.GetDeviceById(deviceThatIsDeletingId, cancellationToken, track: true);
-
-        if (deviceThatIsDeleting.Identity.Address != _userContext.GetAddress())
-            throw new NotFoundException(nameof(deviceThatIsDeleting));
-
         var deviceThatIsBeingDeleted = await _identitiesRepository.GetDeviceById(request.DeviceId, cancellationToken, track: true);
 
-        if (deviceThatIsBeingDeleted.Identity.Address != _userContext.GetAddress())
-            throw new NotFoundException(nameof(deviceThatIsBeingDeleted));
-
-        deviceThatIsBeingDeleted.MarkAsDeleted(deviceThatIsDeletingId);
+        deviceThatIsBeingDeleted.MarkAsDeleted(deviceThatIsDeletingId, deviceThatIsBeingDeleted.Identity.Address);
```

Comment:
This is not correct. here you pass `deviceThatIsBeingDeleted.Identity.Address`. And inside of the method, you check if this parameter matches `this.Identity.Address` => this is always true. You should pass the address from the user context instead.
And most important: I would expect at least 2 tests to fail here:
* The test for the Handler
* The integration test

Why don't they fail?